library/core/src/num/int_macros.rs

```diff
@@ -2068,11 +2068,11 @@ macro_rules! int_impl {
         pub const fn rem_euclid(self, rhs: Self) -> Self {
             let r = self % rhs;
             if r < 0 {
-                if rhs < 0 {
-                    r - rhs
-                } else {
-                    r + rhs
-                }
+                // if rhs is `integer::MIN`, rhs.wrapping_abs() == rhs.wrapping_abs,
```

Comment:
I think this comment needs some work, but it's pretty tricky to explain. This is my best attempt at reworking yours, feel free to change if you prefer.
```
// Semantically equivalent to `if rhs < 0 { r - rhs } else { r + rhs }`.
// If `rhs` is not `Self::MIN`, then `r + abs(rhs)` will not overflow
// and is clearly equivalent, because `r` is negative.
// Otherwise, `rhs` is `Self::MIN`, then we have
// `r.wrapping_add(Self::MIN.wrapping_abs())`, which evaluates
// to `r.wrapping_add(Self::MIN)`, which is equivalent to
// `r - Self::MIN`, which is what we wanted (and will not overflow
// for negative `r`).
```

Reply:
Thanks for your comment, and sorry for my poor English.